waiter/src/waiter/core.clj

```diff
@@ -266,6 +266,46 @@
       (catch Exception e
         (utils/exception->response e request)))))
 
+(defn add-grpc-headers-and-trailers
+  "Finds and attaches the equivalent grpc status codes for the provided http status code."
+  [server-name {:keys [body headers status trailers] :as response}]
+  (if (= server-name (get-in response [:headers "server"]))
```

Comment:
We should definitely make a clear distinction between when Waiter is generating a response and when its a proxied response.  I'm not sure checking the server header is the best way to check for that distinction.  To illustrate my point, a backend should not be able to trigger this code by emitting a `server` header that is the same as Waiter's.

Reply:
I'll add a `:source` key to the responses to track Waiter generated responses.

Reply:
done.